libindy/src/api/did.rs

```diff
@@ -47,29 +51,29 @@ use std::ptr;
 #[no_mangle]
 pub  extern fn indy_create_and_store_my_did(command_handle: i32,
                                             wallet_handle: i32,
-                                            did_json: *const c_char,
+                                            did_info_json: *const c_char,
                                             cb: Option<extern fn(xcommand_handle: i32,
                                                                  err: ErrorCode,
                                                                  did: *const c_char,
                                                                  verkey: *const c_char)>) -> ErrorCode {
-    trace!("indy_create_and_store_my_did: >>> wallet_handle: {:?}, did_json: {:?}", wallet_handle, did_json);
+    trace!("indy_create_and_store_my_did: >>> wallet_handle: {:?}, did_json: {:?}", wallet_handle, did_info_json);
 
-    check_useful_c_str!(did_json, ErrorCode::CommonInvalidParam3);
+    check_useful_json!(did_info_json, ErrorCode::CommonInvalidParam3, my_did_info, MyDidInfo);
```

Comment:
Rust allows re-definition of local variable. So we can use only ```did_info``` name and reassign it to struct after macros call.

Reply:
I'm agree with @AxelNennker that we should drop `_json` suffix here.

If we will use re-definition we should rename origin variables before. Right now only in API  but may be we will use this macro for other places?
@vimmerru do you suggest to rename API parameters to use re-definition?
If we wouldn't like to change API parameters names current solution seems like most correct.

Reply:
I suggest to remove "json" suffix as it looks redundant for me. We have part of cases with suffix and part without suffix and i more like to drop it in all cases as it doesn't provide real benefits. 

In current code we are following approach to avoid creation of new variables in entities conversion chains and it looks more consistent to me. If in the future we will get the case when 2 variables will be needed we can just add optional parameter to the macro.